iothub/service/src/Registry2/ConfigurationsClient.cs

```diff
@@ -0,0 +1,341 @@
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public async virtual Task RemoveAsync(string configurationId, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Removing configuration: {configurationId}", nameof(RemoveAsync));
+
+            try
+            {
+                if (string.IsNullOrWhiteSpace(configurationId))
+                {
+                    throw new ArgumentException(IotHubApiResources.GetString(ApiResources.ParameterCannotBeNullOrWhitespace, "configurationId"));
+                }
+
+                // use wild-card ETag
+                var eTag = new ETagHolder { ETag = "*" };
+                await RemoveAsync(configurationId, eTag, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(RemoveAsync)} threw an exception: {ex}", nameof(RemoveAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Removing configuration: {configurationId}", nameof(RemoveAsync));
+            }
+        }
+
+        /// <summary>
+        /// Deletes a previously registered device from the system.
+        /// </summary>
+        /// <param name="configuration">The configuration being deleted.</param>
+        /// <param name="cancellationToken">The token which allows the operation to be canceled.</param>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
+        public async virtual Task RemoveAsync(Configuration configuration, CancellationToken cancellationToken = default)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Removing configuration: {configuration?.Id}", nameof(RemoveAsync));
+            try
+            {
+                if (string.IsNullOrWhiteSpace(configuration.ETag))
+                    throw new ArgumentException(ApiResources.ETagNotSetWhileDeletingConfiguration);
+                else
+                    await RemoveAsync(configuration.Id, configuration, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(RemoveAsync)} threw an exception: {ex}", nameof(RemoveAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Removing configuration: {configuration?.Id}", nameof(RemoveAsync));
+            }
+        }
+
+        private async Task RemoveAsync(string configurationId, IETagHolder eTagHolder, CancellationToken cancellationToken)
```

Comment:
No need to have a private method for this when the implementation can be moved into the public method. This same pattern can be seen [here](https://github.com/Azure/azure-iot-sdk-csharp/blob/4164acb88e987cfbf92ed3b97c8fd14c17dc6a06/iothub/service/src/Registry2/DevicesClient.cs#L243)